src/index/views/action-Send.vue

```diff
@@ -0,0 +1,392 @@
+        class="info"
+      >
+        <span class="material-icons">info_outline</span>
+        Will send to {{ resolvedKap }}
+      </div>
+    </div>
+    <label>Amount</label>
+    <input
+      v-model="amount"
+      type="number"
+      :min="0"
+      :max="balance"
+      :class="!isAmountValid ? 'invalid' : ''"
+      @input="validateAmountDebounced()"
+    >
+    <div class="row">
+      <a
+        class="balance"
+        @click="setMaxAmount()"
+      >Max: {{ balance }} KOIN</a>
+      <a
+        class="advanced-toggle"
+        @click="toggleAdvanced()"
+      >Advanced
+        <span
+          v-if="!showAdvanced"
+          class="material-icons"
+        >expand_more</span><span
+          v-else
+          class="material-icons"
+        >expand_less</span></a>
+    </div>
+    <div
+      v-if="showAdvanced"
+      class="advanced-content"
+    >
+      <label for="max-mana">Max mana</label>
+      <input
+        v-model="maxMana"
+        type="text"
+      >
+    </div>
+    <div class="actions">
+      <button
+        class="secondary"
+        @click="cancel"
+      >
+        cancel
+      </button>
+      <button
+        :disabled="!isToValidated || !isToValid || !isAmountValid"
+        class="primary"
+        @click="transfer"
+      >
+        transfer
+      </button>
+    </div>
+  </div>
+</template>
+<script>
+import router from "@/index/router";
+import ViewHelper from "@/shared/mixins/ViewHelper";
+import Storage from "@/shared/mixins/Storage";
+import Sandbox from "@/shared/mixins/Sandbox";
+import { Contract, Provider, Signer, utils } from "koilib";
+
+export default {
+  mixins: [Storage, Sandbox, ViewHelper],
+
+  data() {
+    return {
+      address: "loading ",
+      balance: "0",
+      signer: null,
+      provider: null,
+      koinContract: null,
+      koin: null,
+      to: "",
+      amount: "0",
+      mana: "",
+      isToValid: false,
+      isAmountValid: true,
+      isToValidating: false,
+      isToValidated: false,
+      showAdvanced: false,
+      maxMana: 10,
+      resolvedKap: "",
+      network: null
+    };
+  },
+
+  watch: {
+    "$store.state.currentIndexAccount": function () {
+      this.loadAccount(this.$store.state.currentIndexAccount);
+    },
+    "$store.state.currentNetwork": function () {
+      this.loadAccount(this.$store.state.currentIndexAccount);
+    },
+  },
+
+  created() {
+    this.validateToDebounced = this.debounce(this.validateTo);
+    this.validateAmountDebounced = this.debounce(this.validateAmount);
+  },
+
+  mounted() {
+    this.loadAccount(this.$store.state.currentIndexAccount);
+  },
+
+  methods: {
+    setMaxAmount() {
+      this.amount = this.balance;
+    },
+    toggleAdvanced() {
+      this.showAdvanced = !this.showAdvanced;
+    },
+    changeTo() {
+      this.isToValidated = false;
+      this.resolvedKap = "";
+    },
+    async validateTo() {
+      this.isToValidating = true;
+      let isAddress;
+      try {
```

Comment:
do it in this way:
```
if (this.to.includes(".")) {
  // resolve kap name
} else {
  // validate checksum
}
```

Reply:
Resolving per tg convo
![image](https://github.com/joticajulian/kondor/assets/876087/8f77bf47-3f92-4c79-a126-a596496688dc)